hudi-spark/src/main/java/org/apache/hudi/BaseAvroPayload.java

```diff
@@ -30,6 +30,8 @@
  */
 public abstract class BaseAvroPayload implements Serializable {
 
+  protected final GenericRecord genericRecord;
```

Comment:
this is problematic.. Shuffling avro has its own issues.. that's why we only shuffle bytes and deserialize lazily.. Lets keep it that way and remove the member variable